lib/puppet/provider/service/systemd.rb

```diff
@@ -164,10 +164,15 @@ def enable
   end
 
   def mask
-    self.disable
+    disable if exist?
     systemctl_change_enable(:mask)
   end
 
+  def exist?
+    result = execute([command(:systemctl), 'cat', '--', @resource[:name]], :failonfail => false)
+    result.exitstatus == 0
```

Comment:
Maybe a better way is using `systemctl show`? Interestingly enough, it returns something even it doesn't exist:
```console
$ systemctl show --property LoadState puppet.service
LoadState=loaded
$ systemctl show --property LoadState puppet3.service
LoadState=not-found
```